parser.rb

```diff
@@ -0,0 +1,43 @@
+class TextParser
+	attr_reader :text, :word_counts
+	def initialize(file)
+		@text = IO.read(file)
+		@word_counts = {}
+	end
+
+	def remove_newlines
+		text.gsub(/\s+/, ' ').strip
+	end
+
+	def parsed_text
+		remove_newlines.split(" ").map do |string|
+			string.gsub(/[^a-zA-Z0-9'-]/i, '').downcase
+		end
+	end
+
+	def count_words
+		parsed_text.each do |word|
+			if word_counts.key?(word)
```

Comment:
If you instantiate the `word_counts` hash with a default value of `0`, you wouldn't need to check if the word has already been seen.

http://ruby-doc.org/core-2.2.2/Hash.html#method-c-new